float32[16] O_T_EE_array
float32 q7
float32[7] q_actual_array
---
float32[7] q_array